### PR TITLE
[Merged by Bors] - feat(CategoryTheory/Localization): the localized category is preadditive

### PR DESCRIPTION
**Pull request by joelriou**
In this PR, it is shown that if `W : MorphismProperty C` has a left calculus of fractions and `C` is preadditive, then the localized category is also preadditive.

---

- [x] depends on: #11737
- [x] depends on: #11721

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Localization/CalculusOfFractions/Preadditive.lean [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+
+/-- The sum of two left fractions with the same denominator. -/
+@[reducible]
+def add : W.LeftFraction X Y where
+  Y' := φ.Y'
+  f := φ.f + φ.f'
+  s := φ.s
+  hs := φ.hs
+
+lemma symm_add : φ.symm.add = φ.add := by
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Localization/CalculusOfFractions/Preadditive.lean:64]: Is this a useful simp-lemma?
```suggestion
@[simp]
lemma symm_add : φ.symm.add = φ.add := by
```

File: Mathlib/CategoryTheory/Localization/CalculusOfFractions/Preadditive.lean [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+variable (W)
+
+namespace Localization
+
+namespace Preadditive
+
+variable {X Y Z : C}
+variable {L}
+
+/-- The opposite of a map `L.obj X ⟶ L.obj Y` when `L : C ⥤ D` is a localization
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Localization/CalculusOfFractions/Preadditive.lean:82]: I think it makes sense to wrap this whole `neg'` and `add'` section into a `section implementation_details` with a little comment saying that the whole section should never be used directly, but is only there to set up the group structure on the hom-sets.

**Comment by jcommelin** [comment]
LGTM, two minor comments
(Sorry for resolving the merge conflict... I realized too late that this might mess with your multi-stacked-mega-PR-DAG.)

bors d+

**Comment by joelriou** [comment]
Thanks for the review!

bors merge

**Pull request closed without merging**